sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormatWriter.scala

```diff
@@ -150,7 +163,7 @@ object FileFormatWriter extends Logging {
     )
 
     // We should first sort by partition columns, then bucket id, and finally sorting columns.
-    val requiredOrdering = partitionColumns ++ bucketIdExpression ++ sortColumns
+    val requiredOrdering = partitionColumns ++ bucketSortExpression ++ sortColumns
```

Comment:
Once we skip sort on bucket id, the matching actual ordering can be both the ordering with and without bucket id, right?